### PR TITLE
Set KubernetesServiceInstance.secure based on sane defaults

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #272

### CODE REVIEW

**Comment by geoand** [comment]
Only adding "In progress" since I want to rebase the PR onto master once #291 is on

**Comment by geoand** [comment]
@spencergibb I will take care of your comments tomorrow

File: spring-cloud-kubernetes-discovery/src/main/java/org/springframework/cloud/kubernetes/discovery/DefaultIsServicePortSecureResolver.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+ */
+public class DefaultIsServicePortSecureResolver implements IsServicePortSecureResolver {
+
+	private static final Log log = LogFactory.getLog(DefaultIsServicePortSecureResolver.class);
+
+	private static final Set<Integer> KNOWN_SECURE_PORTS = new HashSet<Integer>() {{
+		add(443);
+		add(8443);
+	}};
+	private static final Set<String> TRUTHY_STRINGS = new HashSet<String>() {{
```
[the remark this answers is not in the document]
Comment (geoand) [reply spring-cloud-kubernetes-discovery/src/main/java/org/springframework/cloud/kubernetes/discovery/DefaultIsServicePortSecureResolver.java]: Not a standard really, it's just standard values for labels I have seen people use to indicate the truthiness of something

File: spring-cloud-kubernetes-discovery/src/main/java/org/springframework/cloud/kubernetes/discovery/IsServicePortSecureResolver.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+import java.util.HashMap;
+import java.util.Map;
+
+
+/**
+ * Will be used to determine where a KubernetesServiceInstance is true or not
+ * DefaultIsServicePortSecureResolver is provided as a default if no bean is configured
+ */
+public interface IsServicePortSecureResolver {
```
[the remark this answers is not in the document]
Comment (geoand) [reply spring-cloud-kubernetes-discovery/src/main/java/org/springframework/cloud/kubernetes/discovery/IsServicePortSecureResolver.java]: Fixed

File: spring-cloud-kubernetes-discovery/src/main/java/org/springframework/cloud/kubernetes/discovery/KubernetesDiscoveryProperties.java [hunk 3]
```diff
@@ -34,6 +34,9 @@
 	/** SpEL expression to filter services. */
 	private String filter;
 
+    /** If enabled, all services are considered secured */
+	private boolean secured = false;
```
[the remark this answers is not in the document]
Comment (geoand) [reply spring-cloud-kubernetes-discovery/src/main/java/org/springframework/cloud/kubernetes/discovery/KubernetesDiscoveryProperties.java]: Fixed

File: spring-cloud-kubernetes-discovery/src/main/java/org/springframework/cloud/kubernetes/discovery/DefaultIsServicePortSecureResolver.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+/**
+ * TODO break up into delegates if the implementation get's more complicated
+ * <p>
+ * Returns true if one of the following conditions apply:
+ * <p>
+ * spring.cloud.kubernetes.discovery.secured has been set to true
+ * the service contains a label or an annotation named 'secured' that is truthy
+ * the port is one of the known ports used for secure communication
+ */
+public class DefaultIsServicePortSecureResolver implements IsServicePortSecureResolver {
```
[the remark this answers is not in the document]
Comment (geoand) [reply spring-cloud-kubernetes-discovery/src/main/java/org/springframework/cloud/kubernetes/discovery/DefaultIsServicePortSecureResolver.java]: Interface removed and class made package private

**Comment by spencergibb** [comment]
Squashed and merged with some polish via eb73e05ccb5b805d78b44d45537149979fce0e9a

**Comment by geoand** [comment]
Cool!

**Pull request closed without merging**